src/main.rs

```diff
@@ -47,125 +97,293 @@ impl LogLevel {
 }
 
 #[derive(Subcommand)]
+/// params that greenboot accepts
+///
+/// greenboot health-check -> runs the custom health checks
+///
+/// greenboot rollback -> if boot_counter satisfies it trigger rollback
 enum Commands {
-    Check,
-    Stamp,
-}
-
-#[derive(Serialize, Deserialize, PartialEq, Eq, Hash)]
-struct ServiceStatus {
-    unit: String,
+    HealthCheck,
+    Rollback,
 }
 
-fn check() -> Result<(), Error> {
-    match File::open("/etc/greenboot/upgrade.stamp") {
-        Ok(_) => {
-            log::info!("stamp on disk, removing and running greenboot");
-            std::fs::remove_file("/etc/greenboot/upgrade.stamp")?
+/// runs all the scripts in required.d and wanted.d
+fn run_diagnostics() -> Result<()> {
+    let mut required_script_failure: bool = false;
+    let mut path_exists: bool = false;
+    for path in GREENBOOT_INSTALL_PATHS {
+        let greenboot_required_path = format!("{path}/check/required.d/");
+        if !Path::new(&greenboot_required_path).is_dir() {
+            continue;
```

Comment:
I would log a warning saying that we are skipping the path because it is not a dir